### ISSUE TITLE
The type TypedResults does not exist in the namespace Microsoft.AspNetCore.Http

### ISSUE THREAD

**Issue by JinnFletch** [dotnet/aspnetcore#50527]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

It is either missing or unclear where the type `TypedResults` lives.  Same issue with `HttpResults.Results`, though I assume that's because it's generated by the [ResultsOfTGenerator](https://github.com/dotnet/aspnetcore/blob/main/src/Http/Http.Results/tools/ResultsOfTGenerator/Program.cs)?

Either way, documentation and/or ability need to be exposed on how to use these types within a class library.  They can be used within a Project using the SDK `Microsoft.NET.Sdk.Web`, and of course any project referencing that one for some reason.

But they cannot be used within a Class Library using the FrameworkReference to `Microsoft.AspNetCore.App`.

### Describe the solution you'd like

One solution I may be missing is some other opinionated process or hooks that mimic `TypedResult` such that a particular result's value or state are converted to an `IResult` that will be handled and serialized by the framework as naturally as `TypedResult` is handled.  I currently can't find such hooks.  In my case, using a `Monad.Match` to handle when there is a value, not a value, or possible exception types, and converting that to something like `TypedResult`.

But my biggest hope is that there is simply an oversight on my part and that I CAN use `TypedResult` and `HttpResults.Results` in a class library so I can create my own converter from Monad to TypedResult.  I just don't see how, and I hope `ResultsOfTGenerator` doesn't make it impossible.

**Comment by JinnFletch** [comment]
After having worked on this all day yesterday in my class library and received constant errors such as "`HttpResults` doesn't exist in the namespace `Microsoft.AspNetCore.Http`", and many other issues related to the `Results` and `TypedResults` classes just completely being missing... I tried to do a direct reference to the PACKAGE `Microsoft.AspnetCore.App.Ref`, which is impossible.  But I tried to force it a couple ways and failed.  So I removed the NuGet PackageReference from my csproj... and now the types are showing up.  Even after undoing all git changes and clearing `.vs, bin, obj`... it's working now.  But a literal 8 hour day of trying to get it to work yesterday failed.

So perhaps that can be looked at, but I suppose I don't currently have an issue, assuming my resulting NuGet build is usable by a consumer.  We'll see when that time comes later today.
